### DISCUSSION TITLE
Interactive FAST rendering in Google CoLab

### DISCUSSION THREAD

**Discussion by andreped** [FAST-Imaging/FAST#189] [Ideas]
For demo purposes, it would be great if it was possible to render the `fast.SimpleWindow2D` in Google CoLab.

As I have demontrated in [this gist](https://colab.research.google.com/gist/andreped/b6b0850a6e9349efbbd2501087cc663d/fast-python-opengl.ipynb) it is possible to convert extracted FAST images to numpy and render these using matplotlib, but then it is not possible to perform interactive rendering, like when rendering a WSI.

I have previously used something called `gradio`, which I have used to make demo apps which now are deployed on Hugging Face spaces. It works wonders and also have support for interactive rendering in CoLab. Some examples can be seen in [this gist](https://colab.research.google.com/gist/andreped/57d51f237132ef5b108599906225262e/gradio-model3d-interactive-rendering-demo.ipynb). I don't know how `gradio` did this. Might be that they use JS for the frontend viewer and WebGL to render - at least for the 3D renderer.

I know this is something FAST does not support today, but I believe this could open new open new opportunities, and would make FAST highly relevant for many production pipelines that focus on web-based software solutions.

**Comment by smistad** [comment]
Note to self: This is how they do it: https://github.com/gradio-app/gradio/blob/a4e010a96f1d8a52b3ac645e03fe472b9c3cbbb1/gradio/blocks.py#L2051
